plugins/volume/org.xen.xapi.storage.fsp/sr.py

```diff
@@ -70,6 +107,63 @@ def set_description(self, dbg, sr, new_description):
     def set_name(self, dbg, sr, new_name):
         util.update_sr_metadata(dbg, 'file://' + sr, {'name': new_name})
 
+    def _create_vdi(self, sr, name, path):
+        fsp = importlib.import_module("fsp")
+        cb = fsp.Callbacks()
+        description = ""
+        sharable = False
+        vdi_uuid = str(uuid.uuid4())
+        with VolumeContext(cb, sr, 'w') as opq:
+            image_type = ImageFormat.IMAGE_DIRECTORY
+            image_format = ImageFormat.get_format(image_type)
+            with PollLock(opq, 'gl', cb, 0.5):
+                with cb.db_context(opq) as db:
+                    volume = db.insert_new_volume(0, image_type)
+                    db.insert_vdi(name, description, vdi_uuid, volume.id, sharable)
+                    volume_path = cb.volumeGetPath(opq, str(volume.id))
+                    db.set_vdi_custom_key(vdi_uuid, 'path_to_link', path, True)
+            os.symlink(path, volume_path)
+        return vdi_uuid
+
+    def _check_paths(self, sr):
+        all_vdis = {}
+        fsp = importlib.import_module("fsp")
+        cb = fsp.Callbacks()
+        with VolumeContext(cb, sr, 'w') as opq:
+            with PollLock(opq, 'gl', cb, 0.5):
+                with cb.db_context(opq) as db:
+                    all_vdis = db.get_all_vdi_custom_keys(True)
+        for row in all_vdis:
+            if "path_to_link" in all_vdis[row]:
+                if not os.path.exists(all_vdis[row]["path_to_link"]):
+                    log.debug('Broken symlink {}: uuid={}'.format(all_vdis[row]["path_to_link"], row))
```

Comment:
You can use a temporary variable here:
```python
directory = all_vdis[row]["path_to_link"]
if not os.path.exists(directory):
    log.debug('Broken symlink {}: uuid={}'.format(directory, row))
```
`vdi_uuid` is also more explicit than `row`. :)